### DISCUSSION TITLE
Why getAccounts return array?

### DISCUSSION THREAD

**Discussion by KetulCodiste** [Web3Auth/Web3Auth#556] [Help]
After I am connected using web3Auth, I can get the user wallet address using getAccounts which will return me the array of addresses and the 1st entry in the array will be my wallet address.

My doubt is, why it is returning me an array? As per my observation, it always returns one wallet address in the array. And as I am connected using my google account, this should always return me one wallet address. So, if the array has more than 1 entry, than what are the other entries?

**Comment by yashovardhan** [comment]
Hey @KetulCodiste 

If you have a look at the [JSON RPC API for EVM Blockchains](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_accounts), the function eth_accounts always returns an array of addresses. This same API has been impllemented by [web3.js](https://web3js.readthedocs.io/en/v1.2.11/web3-eth.html#getaccounts) & ethers.js as you can check in their documentations.

A similar question was asked in the [ethereum stack exchange](https://ethereum.stackexchange.com/questions/31967/web3-how-do-i-get-just-the-first-account-on-testrpc-using-web3-eth-getaccounts) as well, which might help you understand the case of this API. Web3Auth directly returns the JSON RPC provider.

**Reply by KetulCodiste** [reply]
web3js is usually used with metamask and other wallet providers which gives the options to connect with the multiple account at the same time. When we connect with metamask, metamask asks us to select the checkboxes. Suppose I have 5 accounts then I would be able to connect with all the 5 accounts at the same time. But in web3Auth, as I am logging in using email address, a single email address will always have a single wallet address, as per my knowledge. Am i right? If yes, then my doubt is clear. If no, means getAccounts can return more than 1 wallet address. So, my doubt is, how an account created with the single email address, can return more than 1 wallet address while using getAccounts() method?

**Reply by SeanLuis** [reply]
@KetulCodiste I have the same doubt, using an external wallet like Metamask if it is possible to obtain all the accounts but in this case using web3auth it is impossible for me to obtain all the accounts, something that I still do not understand.
